File: src/main/java/com/grepp/curdsample/dto/TodayTaskDto.java
package com.grepp.curdsample.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TodayTaskDto {

    List<TaskDto> uncompletedTasks = new ArrayList<>();
    List<TaskDto> completedTasks = new ArrayList<>();

    public static TodayTaskDto from(List<TaskDto> tasks) {
        Map<Boolean, List<TaskDto>> result = tasks.stream()
                .collect(
                        Collectors.partitioningBy(TaskDto::isCompleteStatus)
                );
        return new TodayTaskDto(
                result.get(false),
                result.get(true)
        );
    }

}

File: src/main/java/com/est/crudsample/dto/TodayTaskDto.kt
package com.est.crudsample.dto

data class TodayTaskDto(
    val uncompletedTasks: List<TaskDto>,
    val completedTasks: List<TaskDto>
)

//    companion object {
//        fun from(tasks: MutableList<TaskDto?>): TodayTaskDto {
//            val result = tasks.stream()
//                .collect(
//                    Collectors.partitioningBy(TaskDto::completeStatus)
//                )
//            return TodayTaskDto(
//                result[false],
//                result[true]
//            )
//        }
//    }

fun List<TaskDto>.toTodayTasks() : TodayTaskDto {
    val partition: Pair<List<TaskDto>, List<TaskDto>> = partition { it.completeStatus }
    // completeStatus가 true인 것들, false인 것들을 기준으로 나눠준다.

    val (completed, unCompleted) = partition { it.completeStatus }
    return TodayTaskDto(
        completedTasks = completed,
        uncompletedTasks = unCompleted)
}